File: week_1/exercises/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    float change = 0;
    int coins = 0;
    int cents = 0;

    // Change owed from user
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    cents = round(change * 100);

    // 25 cents
    coins += cents / 25;
    cents -= (cents / 25) * 25;

    // 10 cents
    coins += cents / 10;
    cents -= (cents / 10) * 10;

    // 5 cents
    coins += cents / 5;
    cents -= (cents / 5) * 5;

    // 1 cent
    coins += cents / 1;
    cents -= (cents / 1) * 1;

    // Print the minimum of coins needed to
    // pay the change owed. 
    printf("%d\n", coins);

    return 0;
}

File: week_6/exercises/cash/cash.py
from cs50 import get_float
import math

if __name__ == "__main__":
    # The amount of money of each coin
    coins = [25, 10, 5, 1]
    # The number of coins returned to pay the owed money
    change_coins = 0
    while True:
        change = get_float("Change owed: ")
        # Only accepts positive change
        if change > 0:
            break
    change = round(change * 100)
    for coin in coins:
        # Get the max number of coins that fit into the remaining change
        change_coins += int(change / coin)
        # Update the remaipytning change owed
        change -= int(change / coin) * coin

    print(change_coins)
